src/Client/SceneManager.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+** EPITECH PROJECT, 2023
+** R-Bus
+** File description:
+** SceneManager implementation
+*/
+
+#include "SceneManager.hpp"
+#include "raylib.h"
+#include "ClientSystems.hpp"
+#include "Registry.hpp"
+#include "SystemManagersDirector.hpp"
+
+constexpr int screenWidth  = 1920;
+constexpr int screenHeight = 1080;
+constexpr int fps          = 60;
+
+// NOLINTBEGIN(cppcoreguidelines-avoid-non-const-global-variables)
+bool SceneManager::_init             = false;
+SceneManager SceneManager::_instance = SceneManager();
+// NOLINTEND(cppcoreguidelines-avoid-non-const-global-variables)
+
+SceneManager &SceneManager::getInstance()
+{
+    if (!_init) {
+        _init = true;
+        _instance.init();
+    }
+    return _instance;
+}
+
+SceneManager::SceneManager() : _currentScene(Scene::MAIN_GAME), _stop(false)
+{
+}
+
+static void initRaylib()
+{
+    InitWindow(
+        screenWidth,
+        screenHeight,
+        "raylib [textures] examples - texture source and destination "
+        "rectangles");
```

Comment:
```suggestion
        "R-Bus");
```

en vrai ça serait bien de commencer à mettr ele nom du jeu